core/src/test/java/io/questdb/test/cairo/mv/SampleByRangeCursorTest.java

```diff
@@ -0,0 +1,169 @@
+        Assert.assertEquals(TimestampFormatUtils.parseTimestamp("2024-03-05T00:00:00.000000Z"), cursor.getTimestampHi());
+        Assert.assertFalse(cursor.next());
+    }
+
+    @Test
+    public void testFixedOffset() throws SqlException {
+        final SampleByRangeCursor cursor = new SampleByRangeCursor();
+        final TimestampSampler sampler = TimestampSamplerFactory.getInstance(1, 'd', 0);
+        final long offset = Timestamps.HOUR_MICROS;
+        cursor.of(
+                sampler,
+                null,
+                offset,
+                0,
+                7 * Timestamps.DAY_MICROS - 1,
+                1
+        );
+
+        Assert.assertEquals(offset, cursor.getMinTimestamp());
+        Assert.assertEquals(offset + 7 * Timestamps.DAY_MICROS, cursor.getMaxTimestamp());
+
+        for (int i = 0; i < 7; i++) {
+            Assert.assertTrue(cursor.next());
+            Assert.assertEquals(offset + i * Timestamps.DAY_MICROS, cursor.getTimestampLo());
+            Assert.assertEquals(offset + (i + 1) * Timestamps.DAY_MICROS, cursor.getTimestampHi());
+        }
+        Assert.assertFalse(cursor.next());
+    }
+
+    @Test
+    public void testSmoke() throws SqlException {
+        final SampleByRangeCursor cursor = new SampleByRangeCursor();
+        final TimestampSampler sampler = TimestampSamplerFactory.getInstance(1, 'd', 0);
+        cursor.of(
+                sampler,
+                null,
+                0,
+                0,
+                7 * Timestamps.DAY_MICROS - 1,
+                1
+        );
+
+        Assert.assertEquals(0, cursor.getMinTimestamp());
+        Assert.assertEquals(7 * Timestamps.DAY_MICROS, cursor.getMaxTimestamp());
+
+        for (int i = 0; i < 7; i++) {
+            Assert.assertTrue(cursor.next());
+            Assert.assertEquals(i * Timestamps.DAY_MICROS, cursor.getTimestampLo());
+            Assert.assertEquals((i + 1) * Timestamps.DAY_MICROS, cursor.getTimestampHi());
+        }
+        Assert.assertFalse(cursor.next());
+    }
+
+    @Test
+    public void testTimeZoneWithDst() throws Exception {
+        final SampleByRangeCursor cursor = new SampleByRangeCursor();
+        final TimestampSampler sampler = TimestampSamplerFactory.getInstance(2, 'h', 0);
+        cursor.of(
+                sampler,
+                Timestamps.getTimezoneRules(TimestampFormatUtils.EN_LOCALE, "Europe/Berlin"),
+                0,
+                TimestampFormatUtils.parseTimestamp("2021-03-28T00:59:00.000000Z"),
+                TimestampFormatUtils.parseTimestamp("2021-03-28T08:01:00.000000Z"),
+                2
+        );
+
+        Assert.assertEquals(TimestampFormatUtils.parseTimestamp("2021-03-27T23:00:00.000000Z"), cursor.getMinTimestamp());
+        Assert.assertEquals(TimestampFormatUtils.parseTimestamp("2021-03-28T10:00:00.000000Z"), cursor.getMaxTimestamp());
+
+        // DST edge is here
```

Comment:
Please add tests where the sample by period is smaller than 1 hour around DST edge

Reply:
Added in 8d17c21d7ccd2f9784042da87b26aeed39f4fe4f